Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -621,8 +621,17 @@ void FunctionDefinitionWriter::visit(AST::CallExpression& call)
             return;
         }
 
+        static constexpr std::pair<ComparableASCIILiteral, ASCIILiteral> baseTypesMappings[] {
+            { "f32", "float"_s },
+            { "i32", "int"_s },
+            { "u32", "unsigned"_s }
+        };
```

Comment:
Is this really the right way to do it? Users can create their own types, and their own types will have these kind of constructors too - but we can't literally list the user-defined types inside our WGSL compiler. I'd imagine both the user-defined type constructors and the primitive type constructors would be handled with the same infrastructure.

Reply:
No, this is isn't the right way to do it. But as I mentioned in the changelog, our whole definition of `CallExpression` needs to change, as we currently only allow types as the target, so this will all need to be refactored. I'm thinking that later we'll distinguish whether we're constructing a type or calling a function, and if we're constructing a type we can use the inferred type which would already account for any aliasing.